src/dappsExplorer/DAppsExplorerScreen.tsx

```diff
@@ -0,0 +1,352 @@
+import {
+  RefreshControl,
+  ScrollView,
+  SectionList,
+  SectionListProps,
+  StyleSheet,
+  Text,
+  View,
+} from 'react-native'
+import Animated from 'react-native-reanimated'
+import { SafeAreaView, useSafeAreaInsets } from 'react-native-safe-area-context'
+import { useDispatch, useSelector } from 'react-redux'
+import { DappExplorerEvents } from 'src/analytics/Events'
+import ValoraAnalytics from 'src/analytics/ValoraAnalytics'
+import Touchable from 'src/components/Touchable'
+import {
+  CategoryWithDapps,
+  dappFavoritesEnabledSelector,
+  dappsCategoriesAlphabeticalSelector,
+  dappsListErrorSelector,
+  dappsListLoadingSelector,
+  dappsListSelector,
+  dappsMinimalDisclaimerEnabledSelector,
+  favoriteDappIdsSelector,
+} from 'src/dapps/selectors'
+import { fetchDappsList } from 'src/dapps/slice'
+import { DappFilter, DappSection, DappV1, DappV2 } from 'src/dapps/types'
+import DappCard from 'src/dappsExplorer/DappCard'
+import FavoriteDappsSection from 'src/dappsExplorer/FavoriteDappsSection'
+import { NoResults } from 'src/dappsExplorer/NoResults'
+import useDappFavoritedToast from 'src/dappsExplorer/useDappFavoritedToast'
+import useDappInfoBottomSheet from 'src/dappsExplorer/useDappInfoBottomSheet'
+import useOpenDapp from 'src/dappsExplorer/useOpenDapp'
+import Help from 'src/icons/Help'
+import DrawerTopBar from 'src/navigator/DrawerTopBar'
+import { TopBarIconButton } from 'src/navigator/TopBarButton'
+import colors from 'src/styles/colors'
+import fontStyles from 'src/styles/fonts'
+import { Spacing } from 'src/styles/styles'
+
+const AnimatedSectionList =
+  Animated.createAnimatedComponent<SectionListProps<DappV1 | DappV2, SectionData>>(SectionList)
+
+interface SectionData {
+  data: Array<DappV1 | DappV2>
+  category: CategoryWithDapps
+}
+
+interface DappFilterChip {
+  chipFilter: DappFilter
+  selectedFilter: DappFilter
+  setFilter: (filter: DappFilter) => void
+  lastChip: boolean
+}
+
+export function DAppsExplorerScreen() {
+  const { t } = useTranslation()
+  const insets = useSafeAreaInsets()
+
+  const sectionListRef = useRef<SectionList>(null)
+  const scrollPosition = useRef(new Animated.Value(0)).current
+  const horizontalScrollView = useRef<ScrollView>(null)
+
+  const onScroll = Animated.event([{ nativeEvent: { contentOffset: { y: scrollPosition } } }])
+  const dispatch = useDispatch()
+  const loading = useSelector(dappsListLoadingSelector)
+  const error = useSelector(dappsListErrorSelector)
+  const categories = useSelector(dappsCategoriesAlphabeticalSelector)
+  const dappFavoritesEnabled = useSelector(dappFavoritesEnabledSelector)
+  const dappsMinimalDisclaimerEnabled = useSelector(dappsMinimalDisclaimerEnabledSelector)
+  const dappList = useSelector(dappsListSelector)
+  const favoriteDappsById = useSelector(favoriteDappIdsSelector)
+  const [selectedFilter, setSelectedFilter] = useState({
+    id: 'all',
+    name: t('dappsScreen.allDapps'),
+  })
```

Comment:
we discussed this but can we use just a string value for this, instead of keeping the derived "name" value in state?